docs: Describe when serverpod start launches Docker Compose services (#751)
serverpod/serverpod_docs: docs/11-upgrading/01-upgrade-to-four.md

#### Keep your Docker Postgres (easiest upgrade)

If you've been developing against a Docker Postgres on 3.4, you can keep it without changing your config.

> On projects configured to use a regular PostgreSQL (no `dataPath` in the config), `serverpod start` brings up Docker if it isn't running, and tears down the compose stack on exit if the command brought it up.

Reviewer: Suggestion: this one drops two of the conditions. Auto-start needs a compose file present and a database host of `localhost` or `127.0.0.1`, on top of the missing `dataPath`. "A regular PostgreSQL" would include a remote one, where the stack is deliberately not started.

```suggestion
On projects whose config points at a Postgres on `localhost` with no `dataPath`, `serverpod start` brings up the `docker-compose` stack if it isn't running, and tears it down on exit if the command brought it up.
```
